README.md

```diff
@@ -97,9 +97,7 @@ Optional features:
 
 ## Rust Version Compatibility Policy
 
-DataFusion's Minimum Required Stable Rust Version (MSRV) policy is to support
-each stable Rust version for 6 months after it is
```

Comment:
The previous wording guaranteed support for any rust version for next 6 months.
Given current rust release cadence, shortening the guaranteed support is the main change of this PR. Can we have this reflected in the PR title?

Reply:
Good call -- I changed the title to say

> doc: Update MSRV policy, shortening to max(4 months, 4 releases)